src/mono/mono/metadata/appdomain.c

```diff
@@ -1256,27 +1263,62 @@ mono_runtime_install_appctx_properties (void)
 {
 	ERROR_DECL (error);
 	gpointer args [3];
+	int n_total_props;
+	char **total_keys_raw;
+	char **total_values_raw;
+	gunichar2 **total_keys;
+	gunichar2 **total_values;
 
 	MonoMethod *setup = mono_class_get_method_from_name_checked (mono_class_get_appctx_class (), "Setup", 3, 0, error);
 	g_assert (setup);
 
 	// FIXME: TRUSTED_PLATFORM_ASSEMBLIES is very large
 
+	// Combine and convert properties
+	n_total_props = n_appctx_props + n_runtimeconfig_json_props;
+	total_keys_raw = g_new0 (char*, n_total_props);
+	total_values_raw = g_new0 (char*, n_total_props);;
```

Comment:
Not sure why you need the utf8 versions of the total group of properties.  Just convert from the utf8 ones directly into the utf16 arrays.

Reply:
Fixed. Thanks!